homeassistant/components/hydrawise/sensor.py

```diff
@@ -21,22 +22,106 @@
 from .coordinator import HydrawiseDataUpdateCoordinator
 from .entity import HydrawiseEntity
 
-SENSOR_TYPES: tuple[SensorEntityDescription, ...] = (
-    SensorEntityDescription(
+
+@dataclass(frozen=True, kw_only=True)
+class HydrawiseSensorEntityDescription(SensorEntityDescription):
+    """Describes Hydrawise binary sensor."""
+
+    value_fn: Callable[[HydrawiseSensor], Any]
+
+
+def _get_zone_watering_time(sensor: HydrawiseSensor) -> int:
+    if (current_run := sensor.zone.scheduled_runs.current_run) is not None:
+        return int(current_run.remaining_time.total_seconds() / 60)
+    return 0
+
+
+def _get_zone_next_cycle(sensor: HydrawiseSensor) -> datetime | None:
+    if (next_run := sensor.zone.scheduled_runs.next_run) is not None:
+        return dt_util.as_utc(next_run.start_time)
+    return None
```

Comment:
Euh, as far as I know, if the date time has timezone data, we still convert it to UTC internally, do I'm wondering if this adds any value now

Reply:
It's the other way around. The problem is when the date time does *not* have any timezone data. Then we'll run into:
```
ValueError: Invalid datetime: sensor.raised_beds_next_cycle provides state '2024-05-07 12:55:00', which is missing timezone information
```